hail/build.gradle

```diff
@@ -44,24 +44,30 @@ String hailShortVersion = MAJOR.toString() + "." + MINOR.toString()
 String hailPipVersion = hailShortVersion + "." + PATCH + System.getProperty("hail.pip-version-suffix", "")
 String[] deployedSparkVersions = new File("deployed-spark-versions.txt").readLines()
 
-String sparkVersion = System.getProperty("spark.version", "2.2.0")
+String sparkVersion = System.getProperty("spark.version", "2.4.0")
 String breezeVersion = System.getProperty("breeze.version")
 String py4jVersion = System.getProperty("py4j.version")
 
-if (!(sparkVersion ==~ /^2\..*/))
-    ant.fail('Hail does not support Spark version ' + sparkVersion + '. Hail team recommends version 2.2.0.')
+if (!(sparkVersion ==~ /^2\.[2-4].*/))
+    ant.fail('Hail does not support Spark version ' + sparkVersion + '. Hail team recommends version 2.4.0.')
 
 String scalaVersion = '2.11.8'
 String scalaMajorVersion = '2.11'
 
-if (sparkVersion == '2.2.0') {
-    if (py4jVersion == null)
-      py4jVersion = '0.10.4'
-    if (breezeVersion == null)
-      breezeVersion = '0.13.1'
-} else {
-  if (breezeVersion == null || py4jVersion == null)
-    ant.fail('Unknown Spark version ' + sparkVersion + '. Set breeze.version and py4j.version properties for Spark ' + sparkVersion + '.')
+if (py4jVersion == null) {
+  if (sparkVersion == '2.2.0')
+    py4jVersion = '0.10.4'
+  else if (sparkVersion == '2.3.0')
+    py4jVersion = '0.10.6'
```

Comment:
what about 2.2.1 and 2.3.1 and friends? Should we truncate to the first three characters of sparkVersion for these checks?

Reply:
These are the only ones that are different. Spark 2.2.1 is using 0.10.7, Spark 2.3.1 is using 0.10.7.